server/src/main/java/org/elasticsearch/script/MetricAggScripts.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.elasticsearch.index.fielddata.ScriptDocValues;
+import org.elasticsearch.search.lookup.LeafSearchLookup;
+import org.elasticsearch.search.lookup.SearchLookup;
+
+import java.util.List;
+import java.util.Map;
+
+public class MetricAggScripts {
+    private abstract static class ParamsAndAggBase {
+        private final Map<String, Object> params;
+        private final Object agg;
+
+        ParamsAndAggBase(Map<String, Object> params, Object agg) {
+            this.params = params;
+            this.agg = agg;
+        }
+
+        public Map<String, Object> getParams() {
+            return params;
+        }
+
+        public Object getAgg() {
```

Comment:
Is there any reason that agg should be provided via execute() parameter instead of via an accessor like this?